chunk_downloader.go

```diff
@@ -0,0 +1,533 @@
+// Copyright (c) 2021 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+import (
+	"bufio"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/apache/arrow/go/arrow/ipc"
+	"github.com/apache/arrow/go/arrow/memory"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+)
+
+type snowflakeChunkDownloader struct {
```

Comment:
The interface doesn't specify the member variables and it would result in having to use setters and getters to access or modify them as well as implement multiple functions. The internal fork doesn't take into account the worker based downloader so I think it makes sense to put them together.

Reply:
The internal one used to have both (that's what the interface allowed for). The only reason we deleted the worker based one is because we didn't have good test coverage for it. I'll message you with details.

> The interface doesn't specify the member variables and it would result in having to use setters and getters to access or modify them as well as implement multiple functions. 

All of the methods where you do `if useStreamDownloader()` can be interface methods.